### PR TITLE
Update word2vec model docstring to numpy-style

### PR DESCRIPTION
**Pull request by somnathrakshit**
Changed the docstring of gensim/models/word2vec.py to numpy-style.

### CODE REVIEW

File: gensim/models/word2vec.py [hunk 1]
```diff
@@ -121,13 +121,14 @@
...
-from numpy import exp, dot, zeros, random, dtype, float32 as REAL,\
-    uint32, seterr, array, uint8, vstack, fromstring, sqrt,\
+from numpy import exp, dot, zeros, random, dtype, float32 as REAL, \
+    uint32, seterr, array, uint8, vstack, fromstring, sqrt, \
     empty, sum as np_sum, ones, logaddexp, log, outer
 
 from scipy.special import expit
 
-from gensim import utils, matutils  # utility fnc for pickling, common scipy operations etc
+from gensim import utils, \
```
Comment (menshikh-iv) [inline gensim/models/word2vec.py]: why?

Comment (menshikh-iv) [reply gensim/models/word2vec.py]: We used 120 characters limit (not 80), do not be afraid of long lines

You can check PEP8 with `tox -e flake8` command (with our config for flake8)

File: gensim/models/word2vec.py [hunk 2]
```diff
@@ -136,111 +137,180 @@
...
         Each sentence is a list of string tokens, which are looked up in the model's
         vocab dictionary. Called internally from `Word2Vec.train()`.
         This is the non-optimized, Python version. If you have cython installed, gensim
         will use the optimized version from word2vec_inner instead.
+
+        Parameters
+        ----------
+        model : :class:`~gensim.models.word2vec.Word2Vec`
+            `Word2Vec` instance.
+        sentences : iterable of iterables
```
Comment (menshikh-iv) [inline gensim/models/word2vec.py]: `iterable of list of str`

File: gensim/models/word2vec.py [hunk 3]
```diff
@@ -136,111 +137,180 @@
...
+            Iterable of the sentences directly from disk/network.
+        alpha : float
+            Learning rate.
+        work : :class:`numpy.ndarray`
```
Comment (menshikh-iv) [inline gensim/models/word2vec.py]: please use `numpy.ndarray` without `:class:` for some external links (this doesn't work as link anyway)

File: gensim/models/word2vec.py [hunk 4]
```diff
@@ -136,111 +137,180 @@
...
+        Returns
+        -------
+        int
+            Effective number of words trained.
         """
         result = 0
         for sentence in sentences:
-            word_vocabs = [model.wv.vocab[w] for w in sentence if w in model.wv.vocab and
-                           model.wv.vocab[w].sample_int > model.random.rand() * 2 ** 32]
+            word_vocabs = [model.wv.vocab[w] for w in sentence if
```
Comment (menshikh-iv) [inline gensim/models/word2vec.py]: only hanging indents please (no vertical), here and everywhere, vertical allowed **only** for `def`

File: gensim/models/word2vec.py [hunk 5]
```diff
@@ -1015,27 +1116,32 @@ def __iter__(self):
...
 class LineSentence(object):
-    """Simple format: one sentence = one line; words already preprocessed and separated by whitespace.
-    """
+    """Simple format: one sentence = one line; words already preprocessed and separated by whitespace."""
 
-    def __init__(self, source, max_sentence_length=MAX_WORDS_IN_BATCH, limit=None):
+    def __init__(self, source, max_sentence_length=MAX_WORDS_IN_BATCH,
+                 limit=None):
         """
         `source` can be either a string or a file object. Clip the file to the first
```
Comment (menshikh-iv) [inline gensim/models/word2vec.py]: Many undocumented methods, please document it too.

**Comment by somnathrakshit** [comment]
@menshikh-iv Thanks for your comments. I am working on it. Hope to complete documenting more methods soon. Meanwhile. I shall soon submit another PR that corrects the mistakes in this PR. Please review it.

**Comment by menshikh-iv** [comment]
@somnathrakshit no, please use the same PR for fixing mistakes in w2v (no need to publish many PRs with the exact same goal).

Please ping me when you'll be ready for review again!

**Comment by menshikh-iv** [comment]
Ping @somnathrakshit, when you plan to finish this PR?

**Comment by somnathrakshit** [comment]
@menshikh-iv I have returned the original formatting to the places I didn't touch. As for adding documentation to the undocumented methods, can you please tell me the name of any model that has complete documentation i this repo for reference?

**Comment by menshikh-iv** [comment]
@somnathrakshit for example - https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/models/lsimodel.py

**Comment by menshikh-iv** [comment]
also, pay attention that current PR intersect with #1944

**Comment by somnathrakshit** [comment]
I see that PR #1944 is quite advanced in this work. So, shall I use the work of that PR and try to improve upon it?

**Comment by menshikh-iv** [comment]
@somnathrakshit discuss this question with the author of #1944 @steremma (how you can help him).

**Comment by menshikh-iv** [comment]
Improvement presented in #1944 overlap all current changes, for this reason, I close this PR, anyway, thanks for your work @somnathrakshit :+1:

**Pull request closed without merging**